modules/gdnative/register_types.cpp

```diff
@@ -115,7 +115,125 @@ void actual_discoverer_handler() {
 	ProjectSettings::get_singleton()->save();
 }
 
-GDNativeSingletonDiscover *discoverer = NULL;
+static GDNativeSingletonDiscover *discoverer = NULL;
+
+class GDNativeExportPlugin : public EditorExportPlugin {
+
+protected:
+	virtual void _export_file(const String &p_path, const String &p_type, const Set<String> &p_features);
+};
+
+void GDNativeExportPlugin::_export_file(const String &p_path, const String &p_type, const Set<String> &p_features) {
+	if (p_type != "GDNativeLibrary") {
+		return;
+	}
+
+	Ref<GDNativeLibrary> lib = ResourceLoader::load(p_path);
+
+	if (lib.is_null()) {
+		return;
+	}
+
+	Ref<ConfigFile> config = lib->get_config_file();
+
+	String entry_lib_path;
+	{
+
+		List<String> entry_keys;
+		config->get_section_keys("entry", &entry_keys);
+
+		for (List<String>::Element *E = entry_keys.front(); E; E = E->next()) {
+			String key = E->get();
+
+			Vector<String> tags = key.split(".");
+
+			bool skip = false;
+			for (int i = 0; i < tags.size(); i++) {
+				bool has_feature = p_features.has(tags[i]);
+
+				if (!has_feature) {
+					skip = true;
+					break;
+				}
+			}
+
+			if (skip) {
+				continue;
+			}
+
+			entry_lib_path = config->get_value("entry", key);
+			break;
+		}
+	}
+
+	Vector<String> dependency_paths;
+	{
+
+		List<String> dependency_keys;
+		config->get_section_keys("dependencies", &dependency_keys);
+
+		for (List<String>::Element *E = dependency_keys.front(); E; E = E->next()) {
+			String key = E->get();
+
+			Vector<String> tags = key.split(".");
+
+			bool skip = false;
+			for (int i = 0; i < tags.size(); i++) {
+				bool has_feature = p_features.has(tags[i]);
+
+				if (!has_feature) {
+					skip = true;
+					break;
+				}
+			}
+
+			if (skip) {
+				continue;
+			}
+
+			dependency_paths = config->get_value("dependencies", key);
+			break;
+		}
+	}
+
+	bool is_statically_linked = false;
+	{
+
+		List<String> static_linking_keys;
+		config->get_section_keys("static_linking", &static_linking_keys);
+
+		for (List<String>::Element *E = static_linking_keys.front(); E; E = E->next()) {
+			String key = E->get();
+
+			Vector<String> tags = key.split(".");
+
+			bool skip = false;
+
+			for (int i = 0; i < tags.size(); i++) {
+				bool has_feature = p_features.has(tags[i]);
+
+				if (!has_feature) {
+					skip = true;
+					break;
+				}
+			}
+
+			if (skip) {
+				continue;
+			}
+
+			is_statically_linked = config->get_value("static_linking", key);
+			break;
+		}
+	}
+
+	if (!is_statically_linked)
+		add_shared_object(entry_lib_path);
```

Comment:
We still need the exporter to know about the static library so that the exporter can link it. Also, the export template itself and the exporter code should support linking the static library, so I still think that we should use this behaviour only on iOS and not provide it as an option of GDNativeLibrary.

Reply:
So the exporter should actually *link* a binary? How does that work for cross platform exporting?

Reply:
That's the point - we can't make static linking during export work in a cross-platform way, but we can make iOS specifically perform static linking when exporting. Since dynamic linking is pointless on iOS (you won't be able to publish such bundle in App Store), the idea is to always assume static linking for iOS.

Also, current code in gdnative.cpp is wierd for static linking case - it may omit open_dynamic_library, but then still tries to load symbols as if dlopen was called. What is should do, it call `dlopen(NULL, RTLD_NOW)` and retrieve a handle for the main program and then use `dlsym` on the returned handle.

Reply:
Agree on that, I would not make GDNativeLibrary aware of static linking, since no exporter will be able to perform this if requested save for iOS.